src/Routing/Exception/RedirectException.php

```diff
@@ -27,6 +27,9 @@
  * ```
  * throw new RedirectException('http://example.com/some/path', 301);
  * ```
+ *
+ * If you need a more general purpose redirect exception use
+ * Cake\Http\Exception\RedirectException instead of this class.
```

Comment:
Isn't it confusing to have two identically named `RedirectException` classes?

Reply:
I dont think we need this 2nd exception.
The Error package already heavily relies on the Http one:
- ErrorHandler
- ErrorHandlerMiddleware
- ExceptionRenderer

Reply:
How about we deprecate this exception and make RedirectRoute also throw the new Http/Exception/RedirectException?

Reply:
@dereuromark Using the routing class would make standalone http package usage more cumbersome. I would rather we deprecate the Routing class and have routing depend on Http as that dependency makes more sense.

Reply:
Thats fine :)

Reply:
I've marked the Routing\RedirectException as deprecated, but not changed any of the code as I didn't want to potentially break compatibility over a cosmetic change.